First pointerdown event lost after context menu is triggered

francovs opened the pull request:
#### a19b08297f53de5702b7aa7878843c40bfc08e9e
<pre>
First pointerdown event lost after context menu is triggered
<a href="https://bugs.webkit.org/show_bug.cgi?id=232123">https://bugs.webkit.org/show_bug.cgi?id=232123</a>
<a href="https://rdar.apple.com/84787733">rdar://84787733</a>

Reviewed by Lily Spiniolas and Abrar Rahman Protyasha.

We now explicitly do some cleanup on PointerCaptureController when
showing a contextmenu due right-click. This is required because the
events are not processed while the contextmenu is open, causing the
mousedown that opens the contextmenu to not have a matching mouseup.

The lack of a matching mouseup used to result in one later mousedown
being interpreted as part of a mouse button chord, causing
PointerCaptureController::pointerEventForMouseEvent() to produce a
pointermove instead of a pointerdown.

Canonical link: <a href="https://commits.webkit.org/300696@main">https://commits.webkit.org/300696@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2d1db1d5a9c9d7a1cfd19be42db71772affe6e67

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/123234 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/42948 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/33645 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/129942 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/75347 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/125111 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/43672 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/51543 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/93678 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/62148 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/126187 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/34796 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/110278 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/74305 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/33764 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/28434 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/73457 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/104510 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/28660 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/132659 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/50184 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/38196 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/102168 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/50560 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/106499 "Passed tests") | [❌ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/102024 "Found 1 new API test failure: WebKitGTK/TestInspector:/webkit/WebKitWebInspector/manual-attach-detach (failure)") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/25991 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/47381 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/25601 "Passed tests") | [  ~~🛠 playstation~~](https://ews-build.webkit.org/#/builders/134/builds/46958 "The change is no longer eligible for processing.") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/50039 "Built successfully") | [✅ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/55800 "Built successfully") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/49509 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/52860 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/51188 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/fe29eaab9b30a402fb6d78a580c580eeb3cab9ed. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51164

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/45249f11416276129547f989084ad92db0359441. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51164

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d5d75ab88014c2a2b57fdc458aee1ed3c8e7dad7. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51164

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/02c148d0874607813723b61328388fe786255fb6. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51164

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/be1c28347c46677e7819387ff9a15792395f5425. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51164

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/c3f0f6e7d89a55b2c9b0d4c97ed8ce0d184388ff. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51164

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/681d0cd5e2a62aaadb61a14777cb3462c09db417. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/51164

francovs committed:
First pointerdown event lost after context menu is triggered
https://bugs.webkit.org/show_bug.cgi?id=232123
rdar://84787733

Reviewed by Lily Spiniolas and Abrar Rahman Protyasha.

We now explicitly do some cleanup on PointerCaptureController when
showing a contextmenu due right-click. This is required because the
events are not processed while the contextmenu is open, causing the
mousedown that opens the contextmenu to not have a matching mouseup.

The lack of a matching mouseup used to result in one later mousedown
being interpreted as part of a mouse button chord, causing
PointerCaptureController::pointerEventForMouseEvent() to produce a
pointermove instead of a pointerdown.

Canonical link: https://commits.webkit.org/300696@main
